src/components/SampleConfigProvider.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { PropsWithChildren, useContext, useMemo, useState } from "react";
```

Comment:
@samkim can I get some help reviewing the typescript?

Reply:
yup I'm looking through it